src/pages/instances/actions/snapshots/DuplicateInstanceSnapshotBtn.tsx

```diff
@@ -0,0 +1,48 @@
+import { FC } from "react";
+import { LxdInstance, LxdInstanceSnapshot } from "types/instance";
+import { Button, Icon } from "@canonical/react-components";
+import usePortal from "react-useportal";
+import DuplicateInstanceSnapshotForm from "../../forms/DuplicateInstanceSnapshotForm";
+
+interface Props {
+  instance: LxdInstance;
+  snapshot: LxdInstanceSnapshot;
+  isDeleting: boolean;
+  isRestoring: boolean;
+}
+
+const DuplicateInstanceSnapshotBtn: FC<Props> = ({
+  instance,
+  snapshot,
+  isDeleting,
+  isRestoring,
+}) => {
+  const { openPortal, closePortal, isOpen, Portal } = usePortal();
+
+  return (
+    <>
+      {isOpen && (
+        <Portal>
+          <DuplicateInstanceSnapshotForm
+            close={closePortal}
+            instance={instance}
+            snapshot={snapshot}
+          />
+        </Portal>
+      )}
+      <Button
+        appearance="base"
+        hasIcon
+        dense
+        aria-label="Duplicate instance snapshot"
+        disabled={isDeleting || isRestoring}
+        onClick={openPortal}
+        title="Duplicate instance snapshot"
+      >
+        <Icon name="copy" />
```

Comment:
```suggestion
        <Icon name="plus" />
```
I think the copy icon is more related to copy to clipboard, maybe the plus is better suited?